x-pack/plugins/security_solution/common/api/detection_engine/model/rule_schema_legacy/query_attributes.ts

```diff
@@ -11,24 +11,13 @@ import {
   PositiveIntegerGreaterThanZero,
   enumeration,
 } from '@kbn/securitysolution-io-ts-types';
-
-/**
- * describes how alerts will be generated for documents with missing suppress by fields
- */
-export enum AlertSuppressionMissingFieldsStrategy {
-  // per each document a separate alert will be created
-  DoNotSuppress = 'doNotSuppress',
-  // only alert will be created per suppress by bucket
-  Suppress = 'suppress',
-}
+import { AlertSuppressionMissingFieldsStrategyEnum } from '../rule_schema/specific_attributes/query_attributes.gen';
 
 export type AlertSuppressionMissingFields = t.TypeOf<typeof AlertSuppressionMissingFields>;
 export const AlertSuppressionMissingFields = enumeration(
   'AlertSuppressionMissingFields',
-  AlertSuppressionMissingFieldsStrategy
+  AlertSuppressionMissingFieldsStrategyEnum
```

Comment:
We need to keep the legacy schema and the new schema compatible. The new schema uses string literals instead of Typescript enums. So it is much easier to replace the enum used here with the generated ones instead of adding type castings everywhere these types are used.